schemas/database/001_initial.sql

```diff
@@ -8,3 +8,23 @@ CREATE TABLE transactions (
     raw         BYTEA                    NOT NULL,
     hash        BYTEA                        NULL
 );
+
+CREATE TABLE log_fetches
+(
+    id          SERIAL PRIMARY KEY NOT NULL,
+    timestamp   TIMESTAMP          NOT NULL,
+    first_block BIGINT             NOT NULL,
+    last_block  BIGINT             NOT NULL,
+);
+
+CREATE TABLE logs
+(
+    block_index         BIGINT         NOT NULL,
+    transaction_index   BIGINT         NOT NULL,
+    log_index           BIGINT         NOT NULL,
+    fetch_id            INT            NOT NULL REFERENCES log_fetches (id),
+    group_id            NUMERIC(78, 0) NOT NULL,
+    identity_commitment NUMERIC(78, 0) NOT NULL,
+    root                NUMERIC(78, 0) NOT NULL,
```

Comment:
Instead of storing the decoded event we can also store the `raw` event binary data. This decouples the log-caching a bit more from the event type, making it easier to re-use elsewhere.

To make it fully generic you'd need to solve the dependency on the log filter. This can be done through log-topics, but that is getting quite a bit out of scope so let's leave that for when we need it.